### PR TITLE
ER-1227: Front page feed v3

### PR DESCRIPTION
**Pull request by rimi-itk**
https://jira.itkdev.dk/browse/ER-1227

### CODE REVIEW

File: sites/all/modules/reol_app_feeds/src/Feed/FrontPageFeed.php [hunk 1]
```diff
@@ -194,6 +196,11 @@ protected function getVideos(array $paragraphIds) {
     ], $paragraphIds);
 
     foreach ($paragraphs as $paragraph) {
+      // Don't include video bundles in Front page feed v2.
+      if (self::class === get_class($this)
```
Comment (cableman) [inline sites/all/modules/reol_app_feeds/src/Feed/FrontPageFeed.php:200]: Will `$this` not always be of the type `self:class` or is there some inheritance in play here?  Where this is some type of parent class?

File: sites/all/modules/reol_app_feeds/src/Helper/ParagraphHelper.php [hunk 2]
```diff
@@ -341,12 +341,19 @@ private function getCarousel(\ParagraphsItemEntity $paragraph) {
...
     // Cf. ereol_article_get_articles().
+
     $query = new EntityFieldQuery();
     $count = _reol_app_feeds_variable_get('reol_app_feeds_frontpage', 'max_news_count', 6);
 
+    $bundle = 'article';
+    // Hack for eReolen Go!
+    if (module_exists('breol_news')) {
+      $bundle = 'breol_news';
+    }
```
Comment (cableman) [inline sites/all/modules/reol_app_feeds/src/Helper/ParagraphHelper.php:352]: Did it not see this comment and statement in at PR last week as well? Might need to be moved to a helper function?

File: sites/all/modules/reol_app_feeds/src/Helper/ParagraphHelper.php [hunk 3]
```diff
@@ -745,10 +752,21 @@ private function getVideoBundleVideoList(\ParagraphsItemEntity $paragraph) {
...
 
     $thumbnail = $this->getVideoThumbnail($url);
 
+    $carousels = array_map(static function (array $carousel) {
+      return array_filter($carousel, static function ($key) {
+        return in_array($key, ['query', 'title']);
+      }, ARRAY_FILTER_USE_KEY) + [
+        'type' => 'carousel',
+      ];
+    }, $this->getCarousel($paragraph));
```
Comment (cableman) [inline sites/all/modules/reol_app_feeds/src/Helper/ParagraphHelper.php:768]: Hehehe array magic :-D

**Review by cableman — CHANGES_REQUESTED** [review]

File: sites/all/modules/reol_app_feeds/src/Helper/ParagraphHelper.php (hunk 2 above)
Comment (rimi-itk) [reply sites/all/modules/reol_app_feeds/src/Helper/ParagraphHelper.php:352]: You've seen it in https://github.com/eReolen/base/pull/201/files. I'm working around the fact that ereolen and ereolengo look very similar and yet are not very similar at all. Adding a helper function, `this_is_ereolengo`, say, will not be that useful, and keeping the hack – and the comment – in the code makes it clear that it is actually a hack.

File: sites/all/modules/reol_app_feeds/src/Helper/ParagraphHelper.php (hunk 3 above)
Comment (rimi-itk) [reply sites/all/modules/reol_app_feeds/src/Helper/ParagraphHelper.php:770]: No, it's just programming. I'll add a comment on what's actually going on.

File: sites/all/modules/reol_app_feeds/src/Feed/FrontPageFeed.php (hunk 1 above)
Comment (rimi-itk) [reply sites/all/modules/reol_app_feeds/src/Feed/FrontPageFeed.php:203]: Yes, inheritance is the key here. I've updated the comment to make this more clear. I hope.

**Review by cableman — APPROVED** [review]

**Pull request merged**